gprofiler/kernel_messages.py

```diff
@@ -0,0 +1,52 @@
+import os
+import selectors
+import time
+
+from gprofiler.log import get_logger_adapter
+
+logger = get_logger_adapter(__name__)
+
+
+class DevKmsgReader:
+    def __init__(self):
+        self.dev_kmsg = open("/dev/kmsg")
+        # skip all historical messages:
+        os.lseek(self.dev_kmsg.fileno(), 0, os.SEEK_END)
+        self.selector = selectors.DefaultSelector()
+        self.selector.register(self.dev_kmsg, selectors.EVENT_READ)
+
+    def iter_new_messages(self):
+        messages = []
+        while self.selector.select(0):
+            line = self.dev_kmsg.readline()[:-1]
+            if line[0] == " ":
```

Comment:
Can you give some samples of expected lines here? So the following parsing code makes sense to the reader

Reply:
Added